include/sparrow/dynamic_bitset.hpp

```diff
@@ -380,4 +562,249 @@ namespace sparrow
+     * bitset_reference implementation *
+     ***********************************/
+
+    template <class B>
+    auto bitset_reference<B>::operator=(const self_type& rhs) noexcept -> self_type&
+    {
+        assign(rhs);
+        return *this;
+    }
+
+    template <class B>
+    auto bitset_reference<B>::operator=(self_type&& rhs) noexcept -> self_type&
+    {
+        assign(rhs);
+        return *this;
+    }
+
+    template <class B>
+    auto bitset_reference<B>::operator=(bool rhs) noexcept -> self_type&
+    {
+        assign(rhs);
+        return *this;
+    }
+
+    template <class B>
+    bitset_reference<B>::operator bool() const noexcept
+    {
+        return (m_block & m_mask) != 0;
+    }
+
+    template <class B>
+    bool bitset_reference<B>::operator~() const noexcept
+    {
+        return (m_block & m_mask) == 0;
+    }
+
+    template <class B>
+    auto bitset_reference<B>::operator&=(bool rhs) noexcept -> self_type&
+    {
+        if (!rhs)
+        {
+            reset();
+        }
+        return *this;
+    }
+    
+    template <class B>
+    auto bitset_reference<B>::operator|=(bool rhs) noexcept -> self_type&
+    {
+        if (rhs)
+        {
+            set();
+        }
+        return *this;
+    }
+    
+    template <class B>
+    auto bitset_reference<B>::operator^=(bool rhs) noexcept -> self_type&
+    {
+        if (rhs)
+        {
+            bool old_value = m_block & m_mask;
+            m_block ^= m_mask;
+            m_bitset.update_null_count(old_value, !old_value);
+        }
+        return *this;
+    }
+
+    template <class B>
+    bitset_reference<B>::bitset_reference(bitset_type& bitset, block_type& block, block_type mask)
+        : m_bitset(bitset)
+        , m_block(block)
+        , m_mask(mask)
+    {
+    }
+
+    template <class B>
+    void bitset_reference<B>::assign(bool rhs) noexcept
+    {
+        rhs ? set() : reset();
+    }
+
+    template <class B>
+    void bitset_reference<B>::set() noexcept
+    {
+        bool old_value = m_block & m_mask;
+        m_block |= m_mask;
+        m_bitset.update_null_count(old_value, m_block & m_mask);
+    }
+
+    template <class B>
+    void bitset_reference<B>::reset() noexcept
+    {
+        bool old_value = m_block & m_mask;
+        m_block &= ~m_mask;
+        m_bitset.update_null_count(old_value, m_block & m_mask);
+    }
+
+    template <class B1, class B2>
+    bool operator==(const bitset_reference<B1>& lhs, const bitset_reference<B2>& rhs)
+    {
+        return bool(lhs) == bool(rhs);
+    }
+
+    /**********************************
+     * bitset_iterator implementation *
+     **********************************/
+
+    template <class B, bool is_const>
+    bitset_iterator<B, is_const>::bitset_iterator(bitset_type* bitset, block_type* block, size_type index)
+        : p_bitset(bitset)
+        , p_block(block)
+        , m_index(index)
+    {
+        assert(m_index < bitset_type::s_bits_per_block);
```

Comment:
```suggestion
        assert(0 <= m_index);
        assert(m_index < bitset_type::s_bits_per_block);
```

Reply:
`m_index` cannot be inferior to zero because it's unsigned. Not sure if there is a max value we could check however.